pkg/operator/controller/gatewayclass/servicemeshcontrolplane.go

```diff
@@ -63,12 +60,11 @@ func (r *reconciler) ensureServiceMeshControlPlane(ctx context.Context, gatewayc
 // desiredServiceMeshControlPlane returns the desired servicemeshcontrolplane.
 func desiredServiceMeshControlPlane(name types.NamespacedName, ownerRef metav1.OwnerReference) (*maistrav2.ServiceMeshControlPlane, error) {
 	pilotContainerEnv := map[string]string{
-		"PILOT_ENABLE_GATEWAY_API":                         "true",
-		"PILOT_ENABLE_GATEWAY_API_DEPLOYMENT_CONTROLLER":   "true",
-		"PILOT_ENABLE_GATEWAY_API_STATUS":                  "true",
-		"PILOT_GATEWAY_API_CONTROLLER_NAME":                OpenShiftGatewayClassControllerName,
-		"PILOT_GATEWAY_API_DEFAULT_GATEWAYCLASS":           OpenShiftDefaultGatewayClassName,
-		"PILOT_ENABLE_GATEWAY_API_GATEWAYCLASS_CONTROLLER": "false",
+		"PILOT_ENABLE_GATEWAY_CONTROLLER_MODE":   "true",
+		"PILOT_GATEWAY_API_CONTROLLER_NAME":      OpenShiftGatewayClassControllerName,
+		"PILOT_GATEWAY_API_DEFAULT_GATEWAYCLASS": OpenShiftDefaultGatewayClassName,
+		// Istio will only create the default gateway class if this is true.
+		"PILOT_ENABLE_GATEWAY_API_GATEWAYCLASS_CONTROLLER": "true",
```

Comment:
We don't want Istio to create the default gatewayclass, according to [the workflow that is described in the enhancement proposal](https://github.com/openshift/enhancements/blob/master/enhancements/ingress/gateway-api-with-cluster-ingress-operator.md#workflow-description).  Are we changing the design, or is this change only for testing?

Reply:
Let me look into this some more.  Note that when `PILOT_ENABLE_GATEWAY_API_GATEWAYCLASS_CONTROLLER` is false then istiod also doesn't reconcile the GatewayClass at all.  Jacek found this and discussed in https://redhat-internal.slack.com/archives/C01R9E3SFEV/p1706222283240929?thread_ts=1705959981.087229&cid=C01R9E3SFEV  The default gatewayclass that is created would have  name: openshift-default and controllerName: openshift.io/gateway-controller.